pkg/dns/dns.go

```diff
@@ -0,0 +1,45 @@
+package dns
+
+import (
+	"errors"
+
+	"github.com/openshift/installer/pkg/types/aws"
+	"github.com/openshift/installer/pkg/types/azure"
+	"github.com/openshift/installer/pkg/types/libvirt"
+	"github.com/openshift/installer/pkg/types/none"
+	"github.com/openshift/installer/pkg/types/openstack"
+)
+
+//ConfigProvider is an interface that provides means to fetch the DNS settings
+type ConfigProvider interface {
+	GetBaseDomain() (string, error)
```

Comment:
this is going to be used by `pkg/asset/installconfig` only please move it there.

Reply:
moved